x/auth/tx/service.go

```diff
@@ -41,26 +40,20 @@ func NewTxServer(clientCtx client.Context, simulate baseAppSimulateFn, interface
 	}
 }
 
-var (
-	_ txtypes.ServiceServer = txServer{}
-
-	// EventRegex checks that an event string is formatted with {alphabetic}.{alphabetic}={value}
-	EventRegex = regexp.MustCompile(`^[a-zA-Z_]+\.[a-zA-Z_]+=\S+$`)
-)
-
-const (
-	eventFormat = "{eventType}.{eventAttribute}={value}"
-)
+var _ txtypes.ServiceServer = txServer{}
 
 // GetTxsEvent implements the ServiceServer.TxsByEvents RPC method.
 func (s txServer) GetTxsEvent(ctx context.Context, req *txtypes.GetTxsEventRequest) (*txtypes.GetTxsEventResponse, error) {
```

Comment:
any chance we can move this to the server package? feel like it belongs there?

Reply:
Where exactly?

Reply:
around here? https://github.com/cosmos/cosmos-sdk/blob/main/client/grpc/tmservice/service.go#L249

Reply:
Yeah that's where it should belong. I'll move it.

Reply:
So while I agree that `TxSearch` should exist there, the current gRPC endpoints for `GetTx` and `GetTxEvents` reside under `/cosmos/tx/v1beta1/txs`. Moving these to `clieng/grpc/tmservice` is technically where they belong but that means we would change the query path.

Is this a breaking change we're comfortable with?

cc @aaronc @amaurym

Reply:
Id say we can support both for now, but mark the one in auth as deprecated, then when we refactor auth we remove them.

Reply:
Why would the query path change?

Reply:
Also I think maybe this should be housed in x/tx and wired up by the "tx" module

Reply:
/x/tx could be another place. The reasoning behind sever/tm is because this is logic specific to tendermint. If we have a different consensus engine it is unknown if this api is supported. In the long run, the tendermint specific commands & end points are not native but something that is added to the app through runtime.

Reply:
> Why would the query path change?

`/cosmos/tx/v1beta1/txs` -> `/cosmos/base/tendermint/v1beta1`

> Also I think maybe this should be housed in x/tx and wired up by the "tx" module

These are Tendermint gRPC proxy endpoints, they reside with the others in `tmservice` IMO. Best not create a fractured structure.

Reply:
So maybe they should live in runtime or at least get registered by runtime, which is the module that serves up tm specific things

Reply:
served or plugged into runtime is cool. It wouldn't be a module in the sense of a module today, but a way to extend the server package

Reply:
I don't think `/cosmos/base/tendermint/v1beta1` is a good idea, the idea was to make that service a simple TM proxy. Adding this endpoint there would mean that this service needs to know how to decode txs.

Runtime or x/tx both seems fine to me. I have a preference for x/tx, and we should create a tx module.

Reply:
moving it there for now works, later on when we do a larger refactor of tendermint in the sdk, it will have to be moved to avoid tendermint as a dependency bleeding everywhere

Reply:
And what would the endpoint be? There are currently no queries there.

Reply:
I'd say a new `cosmos/tx/v1` query service.

Reply:
It will rely on the following packages though:

```
	sdk "github.com/cosmos/cosmos-sdk/types"
	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
	"github.com/cosmos/cosmos-sdk/types/query"
	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
```

Is that OK?

Reply:
lets do this another time.